spec/requests/api/v1/users_request_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rails_helper'
+
+RSpec.describe 'USER API' do
+  describe 'happy-paths' do
+    it 'sends a list of users' do
+      create_list(:user, 5)
+
+      get '/api/v1/users'
+
+      expect(response).to be_successful
+
+      json = JSON.parse(response.body, symbolize_names: true)
+      users = json[:data]
+
+      expect(users).to be_an(Array)
+      expect(users.count).to eq(5)
+
+      users.each do |user|
+        expect(user).to have_key(:id)
+        expect(user[:id]).to be_an(String)
+        expect(user).to have_key(:type)
+        expect(user[:type]).to be_an(String)
+        expect(user).to have_key(:attributes)
+        expect(user[:attributes]).to be_a(Hash)
+        expect(user[:attributes]).to have_key(:id)
+        expect(user[:attributes][:id]).to be_an(Integer)
+        expect(user[:attributes]).to have_key(:email)
+        expect(user[:attributes][:email]).to be_an(String)
```

Comment:
We can create a helper method to test the serializer structure and dry up this part of our tests (the same checks are done from lines 42-52). The helper method can be left inside this test or be put in a module and included in the spec helper. For now, I think it's okay to leave it inside this test. It would look something like this:

```
def user_serializer_structure_check(user)
     expect(user).to have_key(:id)
     expect(user[:id]).to be_an(String)
     expect(user).to have_key(:type)
     expect(user[:type]).to be_an(String)
     expect(user).to have_key(:attributes)
    expect(user[:attributes]).to be_a(Hash)
    expect(user[:attributes]).to have_key(:id)
    expect(user[:attributes][:id]).to be_an(Integer)
    expect(user[:attributes]).to have_key(:email)
    expect(user[:attributes][:email]).to be_an(String)
end
```

We could call this whenever we get anything back from the serializer, although I'm not sure if that is redundant. @iandouglas thoughts?